File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        prov = 0
        visited = [0]*len(isConnected)
        for area in range(len(isConnected)) :
            if visited[area] : continue
            visited[area] = 1
            q = deque([area])
            prov+=1
            while q :
                curarea = q.popleft()
                for idx in range(len(isConnected[0])):
                    if curarea != idx and visited[idx] !=1 and isConnected[curarea][idx] == 1 :
                        visited[idx] = 1
                        q.append(idx)
        return prov

File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        int row = isConnected.size();
        int col = isConnected[0].size();
        vector<int> visited(row, 0);
        int prov = 0;
        for (int area=0; area < row; area++) {
            if (visited[area]) continue;
            visited[area] = 1;
            queue<int> Q;
            Q.push(area);
            prov++;
            while (!Q.empty()) {
                int curarea = Q.front(); Q.pop();

                for (int idx=0; idx < row; idx++) {
                    if (curarea != idx && visited[idx] !=1 && isConnected[curarea][idx]==1) {
                        visited[idx] =1;
                        Q.push(idx);
                    }
                }
            }
        }
        return prov;
    }
};